agent/lib_aws_sdk_php.c

```diff
@@ -14,10 +14,420 @@
+    message_params.message_action = NR_SPANKIND_PRODUCER;
+  } else if (command_name_len == AWS_SQS_RECEIVE_MESSAGE_COMMAND_LEN
+             && 0
+                    == nr_strncmp(AWS_SQS_RECEIVE_MESSAGE_COMMAND,
+                                  command_name_string,
+                                  AWS_SQS_RECEIVE_MESSAGE_COMMAND_LEN)) {
+    message_params.message_action = NR_SPANKIND_CONSUMER;
+  } else {
+    /* Nothing to do here so exit. */
+    return;
+  }
+
+  cloud_attrs.aws_operation = command_name_string;
+
+  command_arg_value = nr_lib_aws_sdk_php_get_command_arg_value(
+      AWS_SDK_PHP_SQSCLIENT_QUEUEURL_ARG, NR_EXECUTE_ORIG_ARGS);
+
+  /*
+   *  nr_lib_aws_sdk_php_sqs_parse_queueurl checks for NULL so safe pass
+   * command_arg_value directly in.
+   */
+  nr_lib_aws_sdk_php_sqs_parse_queueurl(command_arg_value, &message_params,
+                                        &cloud_attrs);
+
+  /* Add cloud attributes, if available. */
+
+  nr_segment_traces_add_cloud_attributes(segment, &cloud_attrs);
+
+  /* Now end the instrumented segment as a message segment. */
+  nr_segment_message_end(&segment, &message_params);
+
+  nr_free(command_arg_value);
+
+  /*
+   * These are the message_params params that were strduped in
+   * nr_lib_aws_sdk_php_sqs_parse_queueurl
+   */
+  nr_free(cloud_attrs.cloud_region);
+  nr_free(message_params.destination_name);
+  nr_free(cloud_attrs.cloud_account_id);
+}
+
+void nr_lib_aws_sdk_php_sqs_parse_queueurl(
+    const char* sqs_queueurl,
+    nr_segment_message_params_t* message_params,
+    nr_segment_cloud_attrs_t* cloud_attrs) {
+  char* region = NULL;
+  char* queue_name = NULL;
+  char* account_id = NULL;
+  char queueurl[AWS_QUEUEURL_LEN_MAX];
+  char* queueurl_pointer = NULL;
+
+  if (NULL == sqs_queueurl || NULL == message_params) {
+    return;
+  }
+
+  /*
+   * AWS QueueUrl has a very specific format.
+   * The QueueUrl we are looking for will be of the following format:
+   * queueUrl =
+   * 'https://sqs.REGION_NAME.amazonaws.com/ACCOUNT_ID_NAME/SQS_QUEUE_NAME'
+   * where REGION_NAME, ACCOUNT_ID_NAME, and SQS_QUEUE_NAME are the acutal
+   * values such as: queueUrl =
+   * 'https://sqs.us-east-2.amazonaws.com/123456789012/my_amazing_queue'
+   * If we are unable to match any part of this, the whole decode is suspect and
+   * all values are discarded.
+   *
+   * Due to the overhead involved in escaping the original buffer, creating a
+   * regex, matching a regex, destroying a regex, this method was chosen as a
+   * more performant option because it's a very limited pattern.
+   */
+
+  if (NULL == nr_strlcpy(queueurl, sqs_queueurl, AWS_QUEUEURL_LEN_MAX)) {
+    /* Malformed queueurl, we can't decode this. */
+    return;
+  }
+  queueurl_pointer = queueurl;
+
+  /*
+   * Find the pattern of the AWS queueurl that should immediately precede the
+   * region.
+   */
+  if (0 != strncmp(queueurl_pointer, "https://sqs.", 12)) {
```

Comment:
Based on our conversation - replace hard coded numbers with a `#define` and `sizeof()`.

Reply:
Please see: [09f10e8](https://github.com/newrelic/newrelic-php-agent/pull/1003/commits/09f10e854f9818202179412645b59fc8039b6c27)